eng/pipelines/templates/jobs/ci.tests.yml

```diff
@@ -83,6 +84,21 @@ jobs:
           testRunTitle: "$(OSName) $(TestTargetFramework)"
           testResultsFormat: "VSTest"
           mergeTestResults: true
+      - pwsh: |
+          $dump = Get-ChildItem -Path $(Build.SourcesDirectory) -Include *.dmp -Recurse |Select -First 1
+          if ($dump)
+          {
+            Write-Host "##vso[task.setvariable variable=uploadDump]true"
+            Write-Host "##vso[task.setvariable variable=dumpPath]$dump"
+          }
+        condition: failed()
```

Comment:
Will this always be a failure case? I know some of the ones we were seeing failed but we didn't get a failed step.

Reply:
> Also do we know how largest these dmp files are? Would it make sense to zip them up? For reference the java build is zipping them but I'm not sure if that is necessary or not. https://github.com/Azure/azure-sdk-for-java/blob/main/eng/pipelines/templates/steps/retain-heap-dump-hprofs.yml

They can go up to 1GB.

Reply:
> It is probably worth refactoring this into a template similarly so we can share this and we also need a unique artifact name if we ever need to retry this on a failed build.

Do you have a suggestion for a variable that can be used that would be unique to each retry?

Reply:
See the example at https://github.com/Azure/azure-sdk-for-java/blob/main/eng/pipelines/templates/steps/retain-heap-dump-hprofs.yml#L16 That takes retries into account (System.JobAttempt)

Reply:
Ah wow.. this would have saved me a bunch of time 😆

Reply:
It might even be worth putting this in common - I expect the pattern to always be "given an extension, upload all files of that extension".

Reply:
If we go down that road, why not go ahead and make it an arbitrary Filter pattern :)

Reply:
Thanks @alzimmermsft!